Source/Rules/BufferValidRule.h

```diff
@@ -0,0 +1,25 @@
+#pragma once
+#include "IRule.h"
+
+#include <vector>
+
+namespace MQTT {
+	namespace Rules {
```

Comment:
Shouldnt rules be under protocol?

Reply:
No according to the class diagram it should just be Rules